mesonbuild/backend/ninjabackend.py

```diff
@@ -790,7 +790,7 @@ def generate_vala_compile(self, compiler, target, outfile):
                 vala_input_files.append(s.rel_to_builddir(self.build_to_src))
         if len(src) == 0:
             raise InvalidArguments('Vala library has no Vala source files.')
-        namebase = os.path.splitext(os.path.split(src[0].fname)[1])[0]
+        namebase = target.name
         base_h = namebase + '.h'
```

Comment:
This is problematic because we generally want  the header to be installed in a folder named from the target name (e.g. `foo-1.0/foo.h`), not the file itself. I think that we should strip the `-MAJOR.MINOR` suffix for generating `base_h`.

For other files (GIR, deps, VAPI, ...) that's fine since it's already the convention.